motor_control/throttle/button_control.ino

```diff
@@ -0,0 +1,35 @@
+#include "button_control.h"
+
+ButtonThrottle::ButtonThrottle(const int &pin, int timeToFull): Throttle(pin){
+    this->timeToFull = timeToFull;
+    isPressed = false;
+}
+
+int ButtonThrottle::mappingFunction(int val, std::string mapType){
+  if(mapType == "linear"){
+    return std::min(val/timeToFull, 1) * 1023;
+  }
+  if(mapType == "exponential"){
+  }
+  if(mapType == "logarithmic"){
+  }
+}
+
+/*
+   * Reads the throttle input from analog signal,
+   * Returns 0~1023
+   */
+int ButtonThrottle::compute_motor_value() {
+    if(digitalRead(this->get_pin()) == HIGH){
```

Comment:
You don't need to use digitalRead(). The base class takes care of this for you, and you can directly use `m_raw_value` here.

Reply:
The base class has analogRead() though, not digitalRead - I was thinking of making the base class method that has the analogRead() line a pure virtual one and then overriding it with digitalRead or analogRead as needed, but figured it wasn't worth it for only one command of code

Reply:
Yeah i think the digitalread implementation is good. Yep make the base class one virtual.

Reply:
Yeah I see your point. I think a good compromise is, like explained above, absorbing both analog and digital implementations into the base class to isolate Arduino-specific code as much as we can. This would make our code as platform-independent as possible.